src/main/scala/code/api/util/ApiTag.scala

```diff
@@ -84,7 +77,16 @@ object ApiTag {
   val apiTagStandingOrders = ResourceDocTag("UK-StandingOrders")
   val apiTagStatements = ResourceDocTag("UK-Statements")
   val apiTagTransactions = ResourceDocTag("UK-Transactions")
-  
+
+  private[this] val tagNameSymbolMapTag: MutableMap[String, ResourceDocTag] = MutableMap()
+
+  /**
+    * get a ResourceDocTag by tag symbol string, if not exists, create one with the symbol
+    * @param tagSymbol tag content
+    * @return exists or created ResourceDocTags
+    */
+  def apply(tagSymbol: String): ResourceDocTag =  this.tagNameSymbolMapTag.getOrElseUpdate(tagSymbol, ResourceDocTag(tagSymbol))
```

Comment:
if the same content tag exists, just return it. or not exists create a new one.

Reply:
I'm not sure about this becuase it might lead to a proliferation (many) of tags. If we have to define them as constants, developers will think twice about adding them. But if we can easily add them, then a developer might easily introduce similar tags with different spellings e.g. Account and Accounts.

Reply:
I have removed the follow constants:
  
```
  //Note: the followings are for the code generator -- BerlinGroupV1.3
  val apiTagPaymentInitiationServicePIS = ResourceDocTag("Payment Initiation Service (PIS)")
  val apiTagConfirmationOfFundsServicePIIS = ResourceDocTag("Confirmation of Funds Service (PIIS)")
  val apiTagAccountInformationServiceAIS = ResourceDocTag("Account Information Service (AIS)")
  val apiTagSigningBaskets = ResourceDocTag("Signing Baskets")
  val apiTagCommonServices = ResourceDocTag("Common Services")
```
Because if when we parse new swagger file, we must always add a some tag constants as these constants declaration, this is "hard code". In my opinion for generated api tag, we can call this apply to check whether exists that tag, if not exists, create one return. So for different version api of two swagger files, they can share the same ResourceDocTag instance, because they tag string is the same. that means the tag from swagger file is dynamic, and for developer manually added api tags, they should add constants by manually.

So this apply only for generated apis those from swagger file, please look at the tag in the follow screenshot:
![image](https://user-images.githubusercontent.com/2577334/52615346-e402af80-2ecf-11e9-9548-5c2a20af0875.png)
 So currently, when we generate a group of apis from swagger file, we don't need modify other code, only drop generated code to given package is ok.